### PR TITLE
LPS-130363 Migrate selection modals in commerce-frontend-impl

### PR DESCRIPTION
**Pull request by markocikos**
Hey @liferay-commerce @riccardo-alberti ,

This is a part of [an ongoing epic](https://issues.liferay.com/browse/LPS-129246) to replace legacy AUI-based modals with Clay/React-based `openSelectionModal`. In this PR, we are replacing modals in commerce-frontend-impl.

JIRA: https://issues.liferay.com/browse/LPS-130363

Notes:
- Selection of Tags was broken before this PR, where 'Done' button stays disabled (see gif). As a part of this PR, we are fixing this.
- As a part of this PR, we are also fixing behavior when you unselecting a previously selected category. Before this PR, it used to stay selected. See the correct behavior in the 'after' gif.

Test case:
1. Have a site created using Minium template. This is a prerequisite for all commerce tasks:
   1.   Contol Panel > Sites > (+) Minium
   1.   Contol Panel > Search > Index Actions > Reindex all search indexes.
1. Open mimium site (or place Product Publisher app on a page)
1. Product Publisher > Configuration > Product Selection > Dynamic > Filter > (+). Select a Tag.
1. The same as previous step, Select a Category

Before PR - Tags:
before-select-tag

Before PR - Categories:
before-select-category

After PR:
after

/cc @kresimir-coko @jonmak08

### CODE REVIEW

File: modules/apps/commerce/commerce-frontend-impl/src/main/resources/META-INF/resources/js/liferay_commerce_frontend_asset_categories_selector.js [hunk 1]
```diff
@@ -189,51 +189,50 @@ AUI.add(
...
-											}
-										});
-
-										data[key][0] = key;
-
-										if (!found && !data[key].unchecked) {
-											instance.entries.add(data[key]);
+					const openerWindow = Liferay.Util.getOpener();
+
+					openerWindow.Liferay.Util.openSelectionModal({
```
Comment (kresimir-coko) [inline modules/apps/commerce/commerce-frontend-impl/src/main/resources/META-INF/resources/js/liferay_commerce_frontend_asset_categories_selector.js:194]: How come you had to introduce the `getOpener` to make this work? Just curious 🤔

**Review by kresimir-coko — APPROVED** [review]
LGTM!

**Comment by michaelhashimoto** [comment]
@markocikos - Some of the builds were aborted in this pull request test. Please retest if results needed. Sorry for any inconvenience.

File: modules/apps/commerce/commerce-frontend-impl/src/main/resources/META-INF/resources/js/liferay_commerce_frontend_asset_categories_selector.js (hunk 1 above)
Comment (kresimir-coko) [reply modules/apps/commerce/commerce-frontend-impl/src/main/resources/META-INF/resources/js/liferay_commerce_frontend_asset_categories_selector.js:194]: Sounds good!

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
@FabioDiegoMastrorilli please review this PR!

**Comment by markocikos** [comment]
@FabioDiegoMastrorilli a gentle reminder

**Comment by FabioDiegoMastrorilli** [comment]
ci:forward

**Comment by FabioDiegoMastrorilli** [comment]
Sorry guys, PR makes sense. No changes requested

**Pull request closed without merging**